bin/cli.js

```diff
@@ -1,17 +1,248 @@
 #!/usr/bin/env node
 'use strict';
 
-const metasync = require('metasync');
+const program = require('commander');
+const common = require('metarhia-common');
+const karma = require('karma');
+const path = require('path');
+const fs = require('fs');
 
-const walk = require('../lib/walk');
+const TEST_DIR = 'test';
 
-console.log('Metatest CLI Stub');
+const splitOpts = v => v.split(',');
+
+const cliOptions = [
+  [ '--node',                    'Run in node.js environment' ],
+  [ '--browser',                 'Run in browser environment' ],
+  [ '--browser-only <items>',    'Browser only tests patterns', splitOpts, [] ],
+  [ '--node-only <items>',       'Node only tests patterns', splitOpts, [] ],
+  [ '--exclude <items>',         'Exclude tests patterns', splitOpts, [] ],
+  [ '--browsers <items>',        'Browsers to run', splitOpts, [] ],
+  [ '-l, --browser-log <value>', 'Browser log level' ],
+  [ '-p, --browser-port <n>',    'Browser port' ],
+  [ '-c, --config <value>',      'Config file' ]
+];
+
+const karmaLogLevels = {
+  'disable': [ 'LOG_DISABLE', 'none' ],
+  'error':   [ 'LOG_ERROR', 'errors-only' ],
+  'warn':    [ 'LOG_WARN', 'minimal' ],
+  'info':    [ 'LOG_INFO', 'normal' ],
+  'debug':   [ 'LOG_DEBUG', 'verbose' ]
+};
+
+const browserLaunchers = {
+  'Chrome':    'karma-chrome-launcher',
+  'Firefox':   'karma-firefox-launcher',
+  'IE':        'karma-ie-launcher',
+  'Opera':     'karma-opera-launcher',
+  'PhantomJS': 'karma-phantomjs-launcher',
+  'Safari':    'karma-safari-launcher'
+};
+
+const merge = (arr1 = [], arr2 = []) =>
+  [...common.iter(arr1).join(arr2).collectTo(Set)];
+
+const exclude = (files, filterArr) => {
+  filterArr.map(path => path.replace('.', '[.]')
+    .replace('*', '.+')
+    .replace('?', '.'))
+    .map(path => new RegExp(path))
+    .forEach(regexp => files = files.filter(file => !regexp.test(file)));
```

Comment:
You can use `reduce` for this. 
```js
.reduce((files, regexp) => files.filter(file => !regexp.test(file)), files);
```
and the return the whole chain result.
Also, you could use `common.iter` to avoid creating new array every time, but that's up to you.